recipes/simfil/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
+from conan.tools.files import copy, get
+from conan.tools.scm import Version
+from conan.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.62.0"
+
+
+class SimfilRecipe(ConanFile):
+    name = "simfil"
+    description = "simfil is a C++ 17 library and a language for querying structured map feature data. The library provides an efficient in-memory storage pool for map data, optimized for the simfil query language, along with a query interpreter to query the actual data."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Klebert-Engineering/simfil"
+    license = "BSD 3-Clause"
+    topics = ["query language"]
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_json": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_json": True,
+    }
+
+    exports_sources = "include/*", "src/*"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "10",
+            "clang": "10",
+            "apple-clang": "14",
+        }
+
+    def validate(self):
+        check_min_cppstd(self, 20)
+
+        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warning(f"{self.name} recipe lacks information about the {self.settings.compiler} compiler support.")
+        else:
+            if Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.name} requires Concepts support. The current compiler {self.settings.compiler} {self.settings.compiler.version} does not support it.")
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.28.1")
```

Comment:
```suggestion
        self.tools_requires("cmake/[>3.19 <4]")
```
Cmake is allowed to use range versions: 

https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges

Plus, build_requires is deprecated, tool_requires is its replacement: https://docs.conan.io/1/migrating_to_2.0/recipes.html#requirements